catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala

```diff
@@ -102,27 +97,20 @@ abstract class WorksController[M <: MultipleResultsRequest[W],
     }
   }
 
-  private def getWorkList(request: M, pageSize: Int) = {
-    val works = request.query match {
-      case Some(queryString) =>
-        worksService.searchWorks(
-          queryString,
-          workType = request.workType,
-          pageSize = pageSize,
-          pageNumber = request.page,
-          indexName = request._index
-            .getOrElse(indexName)
-        )
-      case None =>
-        worksService.listWorks(
-          workType = request.workType,
-          pageSize = pageSize,
-          pageNumber = request.page,
-          indexName = request._index
-            .getOrElse(indexName)
-        )
+  private def getWorkList(request: M, pageSize: Int): Future[ResultList] = {
+    val worksSearchOptions = WorksSearchOptions(
+      workTypeFilter = request.workType,
+      indexName = request._index.getOrElse(indexName),
+      pageSize = pageSize,
+      pageNumber = request.page
+    )
+
+    val curriedSearch: (WorksSearchOptions) => Future[ResultList] = request.query match {
```

Comment:
This is a *partial function*, rather than a *partially applied function*.  

To make things more complicated a curried function is a partially applied function, but not all partially applied functions are curried!

- **Currying:** "Decomposition of a function with multiple arguments into a chain of single-argument functions."
- **Partial application of function:** "Passing to a function less arguments than it has in its declaration."
- **Partial function:** "A partial function of type PartialFunction[A, B] is a unary function where the domain does not necessarily include all values of type A."

Reply:
I'd say this is just a function. Since `Function` is a first class type in scala, this is just a function value

Reply:
Actually, why is this a val, rather than a normal def?